python/ql/src/Statements/ReturnOrYieldOutsideOfFunction.qhelp

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>In Python, <code>return</code> and <code>yield</code> statements as well as <code>yield from</code>
+expression can only be used within a function. Having them placed outside of a function or a class
+method will raise a <code>SyntaxError</code> at runtime.</p>
+</overview>
+
+<recommendation>
+<p>The presence of <code>return</code> and <code>yield</code> statements or <code>yield from</code>
+expression outside of a function or a class method suggests a logical error, so it is not possible
```

Comment:
Same as above: either add "the" before "`yield from` expression" or make "expression" plural.

Reply:
Also, I guess the recommendation would be "carefully check the code to see if there are any logical errors". I believe some of our other queries come with this recommendation.

Reply:
I have taken `... suggests a logical error, so it is not possible to suggest a general fix.` from https://help.semmle.com/wiki/display/PYTHON/__init__+method+is+a+generator which I thought made most sense. I don't think suggesting to check the code to find logical errors would make sense as the query already found the logical error. I think the current wording I have would be the most appropriate as it is difficult to suggest how a `return` statement located outside of a function should be fixed. Is it a `return` that was copied-pasted by accident? Did the programmer fail to understand how `return` works? Was it a refactoring issue? Etc.

Reply:
You make a good point. I withdraw my complaint. :slightly_smiling_face:

Reply:
No worres! :)